src/documents/Contexts.md

```diff
@@ -61,15 +52,106 @@ at a particular component location in the tree are the same, as described in the
+which has loaded Infusion - for example, a frame within a browser, or a node.js process. Even when you 
+instantiate apparently "freely-floating" components with a line such as
+
+```
+var that = fluid.myComponent();
+```
+
+these components are also assigned an address in the global tree - they are stored as immediate children
+of the global component root with a uniquely chosen name. Naturally the default visibility rules for
+these components in the component tree have to work differently for these components than they do for
+nested components described by the rules above. If every such component was in scope for every other, there 
+would be "global chaos", to coin a phrase. So all such "free components" are hidden from each other
+through a special exception in the scoping rules.
+
+However, there exist several cases when genuinely global visibility is required for some components - often
+this is accompanied by what is termed a [Singleton Design Pattern](https://en.wikipedia.org/wiki/Singleton_pattern) 
+in prevalent "Design Patterns" dialogue. As commonly described, this pattern consists of two elements - 
+
+1. Arranging for global visibility of a component (object)
+2. Arrange for there to be a single instance of a component
+
+The Infusion framework supplies two special grades, which allow the designer to arrange for both of these
+points to be controlled separately. For a start, any component which is supplied the grade `fluid.resolveRoot`,
+wherever it is constructed in the component tree, will become **_globally visible_** throughout the component
+tree, as if it had been constructed at its root. Secondly, a component which uses the grade 
+`fluid.resolveRootSingle` will be able to arrange that a single instance of it is globally visible at any
+time. Constructing a further instance of such a component will displace the earlier instance from global
+visibility (although it will not destroy it at its original location in the tree). Using the grade `fluid.resolveRootSingle`
+requires an option named `singleRootType` to be defined in the defaults, holding the global type name with 
+respect to which the instance is required to be unique. This needs to be listed explicitly since otherwise the
+framework would not be able to tell which of the component's `gradeNames` or `type` was intended to hold the name of the
+type whose instance was expected to be unique.
+
+For example, the following code is an error, since `that2` cannot see the component `that` with context name `freeComponent1`. 
+
+```javascript
+fluid.defaults("examples.freeComponent1", {
+    gradeNames: ["fluid.component"],
+    freeValue: 42
+});
+
+var that = examples.freeComponent1();
+
+var that2 = fluid.component({
+    value: "{freeComponent1}.options.freeValue" // error: though global, freeComponent1 is not resolvable
+});
+```
+
+However, in the following example, since `rootComponent1` has the grade `fluid.resolveRootSingle`, the instance `root1` can be seen by the
```

Comment:
It might be helpful to the reader to provide an example of how to fix the above scenario – an example that uses the same component names and only adds the minimum of what would be required to allow that2 to see freeComponent1. This would be in addition to the example below.